src/accelerate/data_loader.py

```diff
@@ -326,12 +327,15 @@ class DataLoaderDispatcher(DataLoader):
     """
 
     def __init__(self, dataset, split_batches: bool = False, **kwargs):
+        shuffle = dataset._shuffle_enabled if isinstance(dataset, ShufflerIterDataPipe) else False
```

Comment:
There should be a versioning test of PyTorch first, then this instance check. :-)